Mathlib/RingTheory/Polynomial/Wronskian.lean

```diff
@@ -0,0 +1,116 @@
+/-
+Copyright (c) 2024 Jineon Back and Seewoo Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jineon Baek, Seewoo Lee
+-/
+import Mathlib.Algebra.Polynomial.AlgebraMap
+import Mathlib.Algebra.Polynomial.Derivative
+import Mathlib.LinearAlgebra.SesquilinearForm
+
+/-!
+# Wronskian of a pair of polynomial
+
+This file defines Wronskian of a pair of polynomial, which is `W(a, b) = ab' - a'b`.
+We also prove basic properties of it.
+
+## Main declarations
+
+- `wronskian_eq_sum_zero`: We have `W(a, b) = W(b, c)` when `a + b + c = 0`.
+- `wronskian.degree_lt_add`: Degree of Wronskian `W(a, b)` is strictly smaller than
+  the sum of degrees of `a` and `b`
+- `wronskian.natDegree_lt_add`: `natDegree` version of the above theorem. We need to assume that
+  the Wronskian is nonzero. (Otherwise, `a = b = 1` gives a counterexample.)
+-/
+
+noncomputable section
+
+open scoped Polynomial Classical
+
+namespace Polynomial
+
+variable {R : Type*} [CommRing R]
+
+def wronskian (a b : R[X]) : R[X] :=
+  a * (derivative b) - (derivative a) * b
+
+variable (R) in
+/-- `Polynomial.wronskian` as a bilinear map. -/
+def wronskianBilin : R[X] →ₗ[R] R[X] →ₗ[R] R[X] :=
+  (LinearMap.mul R R[X]).compl₂ derivative - (LinearMap.mul R R[X]).comp derivative
+
+@[simp]
+theorem wronskianBilin_apply (a b : R[X]) : wronskianBilin R a b = wronskian a b := rfl
+
+@[simp]
+theorem wronskian_zero_left (a : R[X]) : wronskian 0 a = 0 := by
+  rw [← wronskianBilin_apply 0 a, map_zero]; rfl
+
+@[simp]
+theorem wronskian_zero_right (a : R[X]) : wronskian a 0 = 0 := (wronskianBilin R a).map_zero
+
+theorem wronskian_neg_left (a b : R[X]) : wronskian (-a) b = -wronskian a b :=
+  LinearMap.map_neg₂ (wronskianBilin R) a b
+
+theorem wronskian_neg_right (a b : R[X]) : wronskian a (-b) = -wronskian a b :=
+  (wronskianBilin R a).map_neg b
+
+theorem wronskian_add_right (a b c : R[X]) : wronskian a (b + c) = wronskian a b + wronskian a c :=
+  (wronskianBilin R a).map_add b c
+
+theorem wronskian_add_left (a b c : R[X]) : wronskian (a + b) c = wronskian a c + wronskian b c :=
+  (wronskianBilin R).map_add₂ a b c
+
+theorem wronskian_is_alt (a : R[X]) : wronskian a a = 0 := by rw [wronskian, mul_comm, sub_self]
+
+theorem IsAltwronskianBilin : (wronskianBilin R).IsAlt := by intro a; exact wronskian_is_alt a
+
+theorem wronskian_self_eq_zero (a : R[X]) : wronskian a a = 0 := wronskian_is_alt a
+
+theorem wronskianBilin_neq_eq (a b : R[X]) : -(wronskianBilin R) a b = (wronskianBilin R) b a :=
+  LinearMap.IsAlt.neg IsAltwronskianBilin a b
```

Comment:
I would recommend dropping this lemma, since it's not the simp-normal form; the one below is all you need, and you can replace `:= wronskianBilin_neq_eq a b` below with this proof.